extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/CdiAwareJob.java

```diff
@@ -34,4 +36,22 @@ public void execute(JobExecutionContext context) throws JobExecutionException {
             }
         }
     }
+
+    @Override
+    public void interrupt() throws UnableToInterruptJobException {
+        Instance.Handle<? extends Job> handle = jobInstance.getHandle();
+        // delegate if possible; throw an exception in other cases
+        if (InterruptableJob.class.isAssignableFrom(handle.getBean().getBeanClass())) {
+            try {
+                ((InterruptableJob) handle.get()).interrupt();
+            } finally {
+                if (handle.getBean().getScope().equals(Dependent.class)) {
+                    handle.destroy();
+                }
```

Comment:
Hi, is it OK to destroy the bean while the quartz working thread is still doing the `Job.execute()`. It will anyway do this in the finally block of the latter.

Reply:
The `destroy()` is a no-op for any other than `@Dependent` beans.

TBF, I am not sure what would be the best way to handle dep. beans here. Currently, the idea is to have `CdiAwareJob create new instance for every invocation of the job (each execute method) and there might be multiple triggers for it.
So at the moment the `interrupt()` implementation also spawns a new instance a handles it separately - hence the destroy logic.

Reply:
> interrupt() implementation also spawns a new instance a handles it separately

I did not notice this. Maybe spawn only one instance inside `execute()`, remember it, and call `interrupt()` on that exactly instance? Consider an example [https://github.com/quartz-scheduler/quartz/blob/main/examples/src/main/java/org/quartz/examples/example7/DumbInterruptableJob.java](https://github.com/quartz-scheduler/quartz/blob/main/examples/src/main/java/org/quartz/examples/example7/DumbInterruptableJob.java). If it  is also a `@Dependent`, the `_interrupted` flag will be set on the instance that is not really executing and checking it.

Reply:
> I did not notice this. Maybe spawn only one instance inside execute(), remember it, and call interrupt() on that exactly instance?

Yea but can't you use the job prescription with several triggers? That would make you keep wrong references.

Reply:
I am not sure that Quartz can `execute()` the same instance of the job class (obtained using JobFactory.newJob()) in different threads simultaneously, unless `JobFactory.newJob()` returns a singleton instance (e.g. @ApplicationScoped).

Consider this answer https://stackoverflow.com/a/10463309
> Quartz creates new instance of your job class every time it wants to trigger that job. 

And also the implementation:
* The instance is stored in [JobRunShell.initialize()](https://github.com/quartz-scheduler/quartz/blob/a5c4d27e963f51097f9b2777489d310a88897ca4/quartz/src/main/java/org/quartz/core/JobRunShell.java#L127)
* which is then passed to [ThreadPool](https://github.com/quartz-scheduler/quartz/blob/a5c4d27e963f51097f9b2777489d310a88897ca4/quartz/src/main/java/org/quartz/core/QuartzSchedulerThread.java#L409)
* which is then passed to [only one](https://github.com/quartz-scheduler/quartz/blob/a5c4d27e963f51097f9b2777489d310a88897ca4/quartz/src/main/java/org/quartz/simpl/SimpleThreadPool.java#L430) worker thread.
* and also there are no other references to instances of Job, and JobRunShell.

Reply:
Hm, yes, I think that we could just keep the `Instance.Handle<? extends Job> handle` reference instead of `Instance<? extends Job> jobInstance`. A job instance should not be shared.

Reply:
This should be resolved via https://github.com/quarkusio/quarkus/pull/42096